Mathlib/Topology/UniformSpace/CompactConvergence.lean

```diff
@@ -318,8 +325,43 @@ theorem tendsto_iff_tendstoUniformly :
   rw [tendsto_iff_forall_isCompact_tendstoUniformlyOn, ← tendstoUniformlyOn_univ]
   exact ⟨fun h => h univ isCompact_univ, fun h K _hK => h.mono (subset_univ K)⟩
 
+open UniformFun in
+/-- When `α` is compact, `f : X → C(α, β)` is continuous if any only if it is continuous when
+reinterpreted as a map `f : X → α →ᵤ β`. -/
+theorem continuous_iff_continuous_uniformFun {X : Type*} [TopologicalSpace X] (f : X → C(α, β)) :
+    Continuous f ↔ Continuous (fun x ↦ ofFun (f x)) := by
+  rw [continuous_iff_continuous_uniformOnFun]
+  exact UniformOnFun.uniformEquivUniformFun β _ isCompact_univ
+    |>.isUniformInducing.isInducing.continuous_iff
+
 end CompactDomain
 
+section ContinuousOnRestrict
+
+/-- Given functions `F i, f` which are continuous on a compact set `s`, `F` tends to `f`
+uniformly on `s` if and only if the restrictions (as elements of `C(s, β)`) converge. -/
+theorem _root_.ContinuousOn.tendsto_restrict_iff_tendstoUniformlyOn {s : Set α} [CompactSpace s]
+    {f : α → β} (hf : ContinuousOn f s) {ι : Type*} {p : Filter ι}
+    {F : ι → α → β} (hF : ∀ i, ContinuousOn (F i) s) :
+    Tendsto (fun i ↦ ⟨_, (hF i).restrict⟩ : ι → C(s, β)) p (𝓝 ⟨_, hf.restrict⟩) ↔
+      TendstoUniformlyOn F f p s := by
```

Comment:
What is the use case for this and the following? I think I understand why working around them would be painful, but it does sound a bit niche.

Reply:
The use case is [here](https://github.com/leanprover-community/mathlib4/pull/24866/files#diff-d5360be9e669b07b8f7e2c7b9fc25d571ee11786582ac3b3e935e59b1c9b49f7R61#L61-62). While I admit it's a bit specialized, I don't think it's too unreasonable, but if you prefer I can inline that proof. (I did use it twice in that file though)

Reply:
Okay this is the kind of things I was expecting. We can keep this lemma then.